pkg/systemstatsmonitor/osfeature_collector.go

```diff
@@ -0,0 +1,108 @@
+package systemstatsmonitor
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	ssmtypes "k8s.io/node-problem-detector/pkg/systemstatsmonitor/types"
+	"k8s.io/node-problem-detector/pkg/util"
+	"k8s.io/node-problem-detector/pkg/util/metrics"
+	"k8s.io/node-problem-detector/pkg/util/metrics/system"
+)
+
+type osFeatureCollector struct {
+	osFeature *metrics.Int64Metric
+}
+
+func NewOsFeatureCollectorOrDie(osFeatureConfig *ssmtypes.OsFeatureStatsConfig) *osFeatureCollector {
+	oc := osFeatureCollector{nil}
+	var err error
+	// Use metrics.Sum aggregation method to ensure the metric is a counter/cumulative metric.
+	if osFeatureConfig.MetricConfigs["system/os_feature"].DisplayName != "" {
+		oc.osFeature, err = metrics.NewInt64Metric(
+			metrics.OsFeatureID,
+			osFeatureConfig.MetricConfigs[string(metrics.OsFeatureID)].DisplayName,
+			"Guest OS Features",
+			"1",
+			metrics.LastValue,
+			[]string{featureLabel, "value"})
+		if err != nil {
+			glog.Fatalf("Error initializing metric for system/os_feature: %v", err)
+		}
+	}
+	return &oc
+}
+
+func (os *osFeatureCollector) recordFeaturesFromCmdline(cmdlineArgs []system.CmdlineArg) {
+	var featuresMap = map[string]int64{
+		"KTD":                    0,
+		"UnifiedCgroupHierarchy": 0,
+		"ModuleSigned":           0,
+		"LoadPinEnabled":         0,
```

Comment:
Should we also consider GPU, and SecureBoot, and CloudProvider?

Reply:
yes we can add.